### PR TITLE
Fix/lp 1929904

### PR DESCRIPTION
**Pull request by ycliuhw**
*Fix juju ssh to a CaaS controller;*

## Checklist

 - [ ] ~Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [ ] ~Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR~
 - [ ] ~Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed~
 - [x] Comments answer the question of why design decisions were made

## QA steps

```console
$ juju bootstrap microk8s k1

$ juju ssh --debug -m k1:controller --container api-server 0

$ juju ssh --debug -m k1:controller --container mongodb 0

$ juju deploy cs:~juju/mariadb-k8s-3

$ juju ssh mariadb-k8s/0
#

```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1929904

### CODE REVIEW

**Comment by ycliuhw** [comment]
[juju ssh on k8s poor ux](https://trello.com/c/R0TEYPbe/1646-juju-ssh-on-k8s-poor-ux)

**Comment by ycliuhw** [comment]
!!build!!

File: cmd/juju/commands/ssh_container.go [hunk 1]
```diff
@@ -187,6 +188,14 @@ func (c *sshContainer) cleanupRun() {
 const charmContainerName = "charm"
 
 func (c *sshContainer) resolveTarget(target string) (*resolvedTarget, error) {
+	if c.modelName == environsbootstrap.ControllerModelName {
+		// TODO(caas): change here to controller unit tag once we refactored controller to an application.
+		if target != "0" {
```
Comment (hpidcock) [inline cmd/juju/commands/ssh_container.go:193]: This breaks applications deployed in the controller model, which should still work.

This change would be sufficient.
```
	if c.modelName == environsbootstrap.ControllerModelName && target == "0" {
		// TODO(caas): change here to controller unit tag once we refactored controller to an application.
		return &resolvedTarget{entity: fmt.Sprintf("%s-%s", environsbootstrap.ControllerModelName, target)}, nil
	}
```

**Review by hpidcock — APPROVED** [review]
Just one fixed needed.

File: cmd/juju/commands/ssh_container.go (hunk 1 above)
Comment (ycliuhw) [reply cmd/juju/commands/ssh_container.go:193]: That's right. Even it's not recommended to deploy a workload to the controller model, we can stop ppl from doing that.

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**